### PR TITLE
pretty-close-to-build-rs

### PR DESCRIPTION
**Pull request by rcoh**
Moves javap into build.rs. Error handling works pretty well. There is one missing piece to also capture `#[java...]` derives
since those must also be reflected in build.rs.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
@rcoh I'm *finally* getting around to reading this -- is it possible you forgot to add `duchess-reflect/src/reflect/javap.rs`?

File: duchess-build-rs/src/derive_java.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use syn::{parse::ParseStream, Attribute};
+
+use crate::re;
+
+pub(crate) fn process_file(
+    rs_file: &crate::files::File,
+    reflector: &mut duchess_reflect::reflect::JavapReflector,
+) -> anyhow::Result<bool> {
+    let mut watch_file = false;
+    eprintln!("Looking for derive(java) in {:?}", rs_file.path);
```
Comment (nikomatsakis) [inline duchess-build-rs/src/derive_java.rs:16]:
```suggestion
```

File: duchess-build-rs/src/derive_java.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    for capture in re::java_derive().captures_iter(&rs_file.contents) {
+        eprintln!("Debug: found derive(java) in {:?}", rs_file.path);
```
Comment (nikomatsakis) [inline duchess-build-rs/src/derive_java.rs:18]:
```suggestion
```

File: duchess-build-rs/src/java_package_macro.rs [hunk 3]
```diff
@@ -1,14 +1,54 @@
...
-use crate::{files::File, java_compiler::JavaCompiler};
+use crate::{files::File, java_package_macro, re};
 
-pub fn process_macro(compiler: &JavaCompiler, file: &File, offset: usize) -> anyhow::Result<()> {
-    let the_impl: JavaPackageMacro = syn::parse_str(file.rust_slice_from(offset))
-        .with_context(|| format!("{} failed to parse java_package macro", file.slug(offset),))?;
+pub fn process_file(rs_file: &File, reflector: &mut JavapReflector) -> anyhow::Result<bool> {
+    let mut watch_file = false;
+    for capture in re::java_package().captures_iter(&rs_file.contents) {
+        eprintln!("Debug: found java macro in {:?}", rs_file.path);
```
Comment (nikomatsakis) [inline duchess-build-rs/src/java_package_macro.rs:10]:
```suggestion
```

**Review by nikomatsakis — APPROVED** [review]
This makes sense to me -- I was removing `eprintln!` statements but I realized maybe you're not quite ready for this to land.

**Comment by rcoh** [comment]
Yeah I have a revision locally that changes them to optional warnings so
you can see them in cargo. Will push that this morning

**Pull request merged**